lib/core/text/icu/caseModification.nit

```diff
@@ -0,0 +1,181 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Adding locale aware case modification to the String and CString classes
+
+module caseModification is pkgconfig("icu-io", "icu-i18n", "icu-uc")
+import locale
+
+`{
+	#include <unicode/utypes.h>
+	#include <unicode/ustring.h>
+
+	#include <string.h>
+	#include <locale.h>
+
+`}
+
+
+
+# There are 3 different types of case modification but they all behave the same way
+# Therefore the main operations are in a parent class
+private abstract class CaseModifier
```

Comment:
The whole architecture with a collection of classes feels heavy to me; since there's just a couple of functions here, I feel they could all be implemented directly in `CString`, and both functions can be regrouped into one as it makes little sense from a high-level perspective to know the size of a string before it's even created.
Or, they can be separate if we want to expose the function to those working directly with CStrings, your choice.

Either way, since CString is fairly low-level and somewhat performance oriented, I'd believe creating a class hierarchy and making a polymorphic call to a low-level C-function is a bit overkill.